### DISCUSSION TITLE
Homebrew JSON API for non-core formulas and casks?

### DISCUSSION THREAD

**Discussion by tddschn** [Homebrew/discussions#2075] [Everyday usage] [answered]
On https://formulae.brew.sh/docs/api/, I noticed that `https://formulae.brew.sh/api/formula.json`, `https://formulae.brew.sh/api/cask.json` and `https://formulae.brew.sh/api/analytics/install/homebrew-core/30d.json` only include the forumlas / casks in the core repository, 

while links on this page https://formulae.brew.sh/analytics/ include stats of those from non-core taps.

Am I missing something or is there a JSON API that I can query to get the install counts / other metadata of non-core formulas and casks??

**Comment by gromgit** [comment, marked as answer]
> is there a JSON API that I can query to get the install counts [...] of non-core formulas and casks??

Yes. You missed [this section](https://formulae.brew.sh/docs/api/#list-one-category-of-analytics-events) of the docs:
```
$ curl -sL https://formulae.brew.sh/api/analytics/install/30d.json | jq -r '.items[].formula | select(startswith("gromgit"))'
gromgit/fuse/sshfs-mac
gromgit/fuse/ntfs-3g-mac
[...]
```
> is there a JSON API that I can query to get [...] other metadata of non-core formulas and casks??

Not that I know of. Metadata for core formulae/casks can be updated with each PR; gathering metadata for non-core formulae/casks simply can't scale.

**Reply by ipatch** [reply]
> Not that I know of. Metadata for core formulae/casks can be updated with each PR; gathering metadata for non-core formulae/casks simply can't scale.

that's a shame
